File: serialization/typescript/text/src/textSerializationWriterFactory.ts
import {
  SerializationWriter,
  SerializationWriterFactory,
} from "@microsoft/kiota-abstractions";

import { TextSerializationWriter } from "./textSerializationWriter";

export class TextSerializationWriterFactory
  implements SerializationWriterFactory {
  public getValidContentType(): string {
    return "text/plain";
  }
  public getSerializationWriter(contentType: string): SerializationWriter {
    if (!contentType) {
      throw new Error("content type cannot be undefined or empty");
    } else if (this.getValidContentType() !== contentType) {
      throw new Error(`expected a ${this.getValidContentType()} content type`);
    }
    return new TextSerializationWriter();
  }
}

File: serialization/dotnet/text/src/TextSerializationWriterFactory.cs
using System;
using Microsoft.Kiota.Abstractions.Serialization;

namespace Microsoft.Kiota.Serialization.Text;

/// <summary>
/// The <see cref="ISerializationWriterFactory"/> implementation for the text content type
/// </summary>
public class TextSerializationWriterFactory : ISerializationWriterFactory {
    /// <inheritdoc />
    public string ValidContentType { get; } = "text/plain";

    /// <inheritdoc />
    public ISerializationWriter GetSerializationWriter(string contentType)
    {
        if(string.IsNullOrEmpty(contentType))
            throw new ArgumentNullException(nameof(contentType));
        else if(!ValidContentType.Equals(contentType, StringComparison.OrdinalIgnoreCase))
            throw new ArgumentOutOfRangeException($"expected a {ValidContentType} content type");

        return new TextSerializationWriter();
    }
}
